Sources/CorePayments/Networking/APIClient.swift

```diff
@@ -78,12 +79,15 @@ public class APIClient {
     
     // MARK: - Private Methods
     
-    private func constructRESTURL(path: String, queryParameters: [String: String]) throws -> URL {
+    private func constructRESTURL(path: String, queryParameters: [String: String]?) throws -> URL {
         let urlString = coreConfig.environment.baseURL.appendingPathComponent(path)
         var urlComponents = URLComponents(url: urlString, resolvingAgainstBaseURL: false)
         
-        queryParameters.forEach {
-            urlComponents?.queryItems?.append(URLQueryItem(name: $0.key, value: $0.value))
+        if let queryParameters {
+            urlComponents?.queryItems = []
```

Comment:
Do we need this line? I thought it'd just create it for us in the append below if needed.

Reply:
Writing the unit tests for this function revealed that the previous implementation wasn't appending the query parameters at all before. If we didn't set to an empty array, `urlComponents.queryItems` was `nil` and the line in the `forEach` did nothing due to nil coalescing.

Reply:
None of our APIs currently set any query params, which is why our tests prior never caught this. But now that we're breaking this up and writing more specific unit tests we were able to see it not working.